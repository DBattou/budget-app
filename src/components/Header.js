import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
  <header>
    <h1>BudgetApp</h1>
    <ul>
      <li>
        <NavLink to="/" activeClassName="is-active" exact={true}>
          Dashboard
        </NavLink>
      </li>
      <li>
        <NavLink to="/help" activeClassName="is-active">
          Help page
        </NavLink>
      </li>
      <li>
        <NavLink to="/add" activeClassName="is-active">
          Add expense
        </NavLink>
      </li>
    </ul>
  </header>
);

export default Header;
