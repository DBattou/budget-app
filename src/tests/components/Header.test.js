import React from "react";
import { shallow } from "enzyme";
// import ToJson from "enzyme-to-json"; => already imported in jest.config.json
import Header from "../../components/Header";

test("should render header correctly", () => {
  const wrapper = shallow(<Header />);

  expect(ToJson(wrapper)).toMatchSnapshot();

  // expect(ToJson(wrapper)).toMatchSnapshot(); => The usage of we dont want to set it up in jest.config.json
  // expect(wrapper.find("h1").text()).toBe("BudgetApp"); => example of how we can use enzyme
  // More informations https://airbnb.io/enzyme/docs/api/ReactWrapper/find.html
});
