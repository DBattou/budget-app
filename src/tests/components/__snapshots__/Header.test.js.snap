// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    BudgetApp
  </h1>
  <ul>
    <li>
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        exact={true}
        to="/"
      >
        Dashboard
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        to="/help"
      >
        Help page
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        to="/add"
      >
        Add expense
      </NavLink>
    </li>
  </ul>
</header>
`;
